gateway/Dockerfile

```diff
@@ -4,8 +4,7 @@ ENV GOOS=linux
 ENV GOARCH=amd64
 WORKDIR /src
 COPY . .
-RUN go mod download
-RUN go build -o /gateway
+RUN go mod download && go build -o /gateway
```

Comment:
usually better to keep them in dedicated layers, so downlaod cache can be reused if you need to rebuild

but since it's a demo project, it's not a big deal